src/d3a/models/area/event_dispatcher.py

```diff
@@ -75,9 +75,12 @@ def _broadcast_notification(self, event_type: Union[MarketEvent, AreaEvent], **k
 
             if not self.area.events.is_connected:
                 break
+            # TODO: Replace agents with just one agent, since one object is
+            # there fore every market slot
```

Comment:
Yes! Shall we create a story for this?

Reply:
Thanks for pointing out, I should delete this comment. There are unfortunately more than one agents even now, in order to accommodate the future markets, therefore we cannot do that unfortunately.